src/lib/support/tests/TestTestPersistentStorageDelegate.cpp

```diff
@@ -107,66 +109,68 @@ void TestBasicApi(nlTestSuite * inSuite, void * inContext)
 
     // Read them back
 
+    uint8_t all_zeroes[sizeof(buf)];
+    memset(&all_zeroes[0], 0, sizeof(all_zeroes));
+
     memset(&buf[0], 0, sizeof(buf));
-    size = sizeof(buf);
+    size = actualSizeOfBuf;
     err  = storage.SyncGetKeyValue("key2", &buf[0], size);
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
     NL_TEST_ASSERT(inSuite, size == strlen(kStringValue2));
+    NL_TEST_ASSERT(inSuite, size < actualSizeOfBuf);
     NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[0], kStringValue2, strlen(kStringValue2)));
+    // Make sure that there was no buffer overflow during SyncGetKeyValue
+    NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[size], &all_zeroes[0], sizeof(buf) - size));
 
     memset(&buf[0], 0, sizeof(buf));
-    size = sizeof(buf);
+    size = actualSizeOfBuf;
     err  = storage.SyncGetKeyValue("key3", &buf[0], size);
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
     NL_TEST_ASSERT(inSuite, size == strlen(kStringValue3));
+    NL_TEST_ASSERT(inSuite, size < actualSizeOfBuf);
     NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[0], kStringValue3, strlen(kStringValue3)));
+    // Make sure that there was no buffer overflow during SyncGetKeyValue
+    NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[size], &all_zeroes[0], sizeof(buf) - size));
 
+    // Read providing too small a buffer. Data read up to `size` and nothing more.
     memset(&buf[0], 0, sizeof(buf));
-    size = sizeof(buf);
-    err  = storage.SyncGetKeyValue("key2", &buf[0], size);
-    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
-    NL_TEST_ASSERT(inSuite, size == strlen(kStringValue2));
-    NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[0], kStringValue2, strlen(kStringValue2)));
-
-    // Pre-clear buffer to make sure next operations don't change contents
-    uint8_t all_zeroes[sizeof(buf)];
-    memset(&buf[0], 0, sizeof(buf));
-    memset(&all_zeroes[0], 0, sizeof(all_zeroes));
-
-    // Read in too small a buffer: no data read, but correct size given
-    memset(&buf[0], 0, sizeof(buf));
-    size = static_cast<uint16_t>(strlen(kStringValue2) - 1);
-    err  = storage.SyncGetKeyValue("key2", &buf[0], size);
+    size                               = static_cast<uint16_t>(strlen(kStringValue2) - 1);
+    uint16_t sizeBeforeGetKeyValueCall = size;
+    err                                = storage.SyncGetKeyValue("key2", &buf[0], size);
     NL_TEST_ASSERT(inSuite, err == CHIP_ERROR_BUFFER_TOO_SMALL);
-    NL_TEST_ASSERT(inSuite, size == strlen(kStringValue2));
-    NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[0], &all_zeroes[0], sizeof(buf)));
+    NL_TEST_ASSERT(inSuite, size != strlen(kStringValue2));
```

Comment:
```suggestion
    NL_TEST_ASSERT(inSuite, size == sizeBeforeGetKeyValueCall);
```

Reply:
Why, I think this is a valid check

Reply:
It's not "invalid" but the usage of it when we have an explicit size check right after doesn't seem to add significant value

Reply:
I was trying to create a way that if someone changed kStringValue2 they can easily see that it breaks other tests. I will just remove this line.